logdetective/server/database/models/merge_requests.py

```diff
@@ -0,0 +1,396 @@
+import datetime
+from typing import Optional
+
+import backoff
+
+from sqlalchemy import (
+    Column,
+    Integer,
+    DateTime,
+    String,
+    ForeignKey,
+    UniqueConstraint,
+    desc,
+)
+from sqlalchemy.orm import relationship
+from sqlalchemy.exc import OperationalError
+from logdetective.server.database.base import Base, transaction, DB_MAX_RETRIES
+
+
+class MergeRequests(Base):
```

Comment:
We should probably call this `GitlabMergeRequests` since the relationships here are GItlab-specific. There will likely be significant differences in the way that other forges handle things that we may need to account for in the future.